crates/ruff_python_semantic/src/analyze/class.rs

```diff
@@ -170,3 +171,51 @@ pub fn is_metaclass(class_def: &ast::StmtClassDef, semantic: &SemanticModel) ->
         (false, _) => IsMetaclass::No,
     }
 }
+
+/// Returns true if a class is generic.
+///
+/// A class is generic if at least one of its direct bases is subscripted with a `TypeVar`-like,
+/// or if it is defined using PEP 695 syntax.
+///
+/// This should only be used in stub context to avoid false positives and negatives.
```

Comment:
I think even in stub contexts, this could lead to false positives where we say conclude that a class is not generic even though it actually is, and then incorrectly apply an autofix based on that conclusion. That's because the `is_old_style_typevar_like` function doesn't recognise typevars declared in other files as being typevars: I don't think it would understand `typing.AnyStr`, for example. It's pretty rare in typeshed for stubs to use typevars imported from other files (other than `AnyStr`), but I've seen it done much more in other stubs. I definitely don't think we can safely assume that a symbol isn't a typevar just because it's been imported from another module.

Rather than `is_old_style_typevar_like` and `is_generic` functions, I think we actually need `could_be_old_style_typevar_like` and `could_be_generic` functions. `could_be_old_style_typevar_like` would return `true` if it can trace the binding to a TypeVar declaration in the same module, but it would also return `true` if the symbol comes from any other module. `could_be_generic` would have exactly the same implementation you have now, but it would use `could_be_old_style_typevar_like` instead of `is_typevar_like`.